test/dotnet-project-convert.Tests/DotnetProjectConvertTests.cs

```diff
@@ -205,8 +243,353 @@ public void NestedDirectory()
+        File.WriteAllText(Path.Join(testInstance.Path, "Program.cs"), """
+            #:sdk Aspire.Hosting.Sdk/9.1.0
+            Console.WriteLine();
+            """);
+
+        new DotnetCommand(Log, "project", "convert", "Program.cs")
+            .WithWorkingDirectory(testInstance.Path)
+            .Execute()
+            .Should().Pass();
+
+        new DirectoryInfo(testInstance.Path)
+            .EnumerateFileSystemInfos().Select(f => f.Name).Order()
+            .Should().BeEquivalentTo(["Program"]);
+
+        new DirectoryInfo(Path.Join(testInstance.Path, "Program"))
+            .EnumerateFileSystemInfos().Select(f => f.Name).Order()
+            .Should().BeEquivalentTo(["Program.csproj", "Program.cs"]);
+
+        File.ReadAllText(Path.Join(testInstance.Path, "Program", "Program.cs"))
+            .Should().Be("Console.WriteLine();");
+
+        File.ReadAllText(Path.Join(testInstance.Path, "Program", "Program.csproj"))
+            .Should().Be("""
+                <Project Sdk="Aspire.Hosting.Sdk/9.1.0">
+
+                  <PropertyGroup>
+                    <OutputType>Exe</OutputType>
+                    <TargetFramework>net10.0</TargetFramework>
+                    <ImplicitUsings>enable</ImplicitUsings>
+                    <Nullable>enable</Nullable>
+                  </PropertyGroup>
+
+                </Project>
+
+                """);
+    }
+
+    [Fact]
+    public void Directives()
+    {
+        VerifyConversion(
+            inputCSharp: """
+                #!/program
+                #:sdk Microsoft.NET.Sdk
+                #:sdk Aspire.Hosting.Sdk/9.1.0
+                #:property TargetFramework=net11.0
+                #:package System.CommandLine=2.0.0-beta4.22272.1
+                #:property LangVersion=preview
+                Console.WriteLine();
+                """,
+            expectedProject: """
+                <Project Sdk="Microsoft.NET.Sdk">
+
+                  <Sdk Name="Aspire.Hosting.Sdk" Version="9.1.0" />
+
+                  <PropertyGroup>
+                    <OutputType>Exe</OutputType>
+                    <TargetFramework>net10.0</TargetFramework>
+                    <ImplicitUsings>enable</ImplicitUsings>
+                    <Nullable>enable</Nullable>
+                  </PropertyGroup>
+
+                  <PropertyGroup>
+                    <TargetFramework>net11.0</TargetFramework>
+                    <LangVersion>preview</LangVersion>
+                  </PropertyGroup>
+
+                  <ItemGroup>
+                    <PackageReference Include="System.CommandLine" Version="2.0.0-beta4.22272.1" />
+                  </ItemGroup>
+
+                </Project>
+
+                """,
+            expectedCSharp: """
+                Console.WriteLine();
+                """);
+    }
+
+    [Fact]
+    public void Directives_Variable()
+    {
+        VerifyConversion(
+            inputCSharp: """
+                #:package MyPackage=$(MyProp)
```

Comment:
It will be good to have a test at some point which shows that `#:package Package` by itself works, when a `Directory.Versions.props` (or whatever the right conventional file is) is adding a `<PackageVersion Include="Package" Version="..." />` item.

Reply:
I can add such test now, good idea, thanks.